docs/source/release_notes/release-notes.rst

```diff
@@ -43,6 +43,11 @@ develop
          - Change
 
     *    - |fixed|
+         - ``PaxosTimestampBoundStore`` now throws ``NotCurrentLeaderException``, invalidating the timestamp store, if a bound update fails because another timestamp service on the same node proposed a smaller bound for the same sequence number.
```

Comment:
This PR is aimed at `develop`, and so I'll add this note to both develop and 0.42.1, and add a 0.42.0 header too